common/lib/xmodule/xmodule/word_cloud_module.py

```diff
@@ -237,23 +234,38 @@ def handle_ajax(self, dispatch, data):
                 'error': 'Unknown Command!'
             })
 
-    def get_html(self):
-        """Template rendering."""
-        context = {
+    def student_view(self, context):
+        """
+        Renders the output that a student will see.
+        """
+        fragment = Fragment()
+
+        fragment.add_content(self.system.render_template('word_cloud.html', {
             'ajax_url': self.system.ajax_url,
             'display_name': self.display_name,
             'instructions': self.instructions,
             'element_class': self.location.category,
             'element_id': self.location.html_id(),
             'num_inputs': self.num_inputs,
             'submitted': self.submitted,
-        }
-        self.content = self.system.render_template('word_cloud.html', context)
-        return self.content
+        }))
+
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/d3.min.js'))
```

Comment:
From looking at the sandbox, I see that all four of these files are now loaded individually. I wonder if you should create a bundle for them, although maybe D3 deserves to be separate in case we use it elsewhere.

If you want to do this, you should probably look at how Eugeny implemented the Discussion XBlock. It loads the resources individually in development mode, but knows how to load the bundle when in production:

https://github.com/edx/edx-platform/blob/master/openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py#L133